src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.mapper.impl.response.ShoppingCartResponseMapper;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.model.User;
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private ShoppingCartService shoppingCartService;
+    private MovieSessionService movieSessionService;
+    private UserService userService;
+    private ShoppingCartResponseMapper shoppingCartResponseMapper;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  MovieSessionService movieSessionService,
+                                  UserService userService,
+                                  ShoppingCartResponseMapper shoppingCartResponseMapper) {
+        this.shoppingCartService = shoppingCartService;
+        this.movieSessionService = movieSessionService;
+        this.userService = userService;
+        this.shoppingCartResponseMapper = shoppingCartResponseMapper;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
+        return shoppingCartResponseMapper.toDto(
+                shoppingCartService.getByUser(userService.get(userId)));
+    }
+
+    @PutMapping("/movie-sessions")
+    public ShoppingCartResponseDto update(@RequestParam Long userId,
+                                          @RequestParam Long movieSessionId) {
+        User user = userService.get(userId);
+        MovieSession movieSession = movieSessionService.get(movieSessionId);
+        shoppingCartService.addSession(movieSession, user);
+        return shoppingCartResponseMapper.toDto(
+                shoppingCartService.getByUser(userService.get(userId)));
```

Comment:
You have found a user already, why do you need to make a request to the DB one more time?